registry/registry/ansible/play.yml

```diff
@@ -0,0 +1,155 @@
+        owner: ubuntu
+        group: ubuntu
+        mode: "g=rwx,o=rwx"
+    - name: add docker repo key with apt
+      become: yes
+      apt_key:
+        id: "{{ docker_apt_key_id }}"
+        keyserver: "{{ docker_apt_key_url }}"
+        state: present
+    - name: add docker repository to sources.list
+      shell:
+        cmd: echo "deb https://apt.dockerproject.org/repo ubuntu-xenial main" | sudo tee /etc/apt/sources.list.d/docker.list
+    - name: update apt cache
+      become: yes
+      apt:
+        update_cache: yes
+    - name: upgrade to latest
+      become: yes
+      apt:
+        upgrade: safe
+    - name: update kernel
+      raw: "sudo apt-get install -qq linux-image-extra-$(uname -r) linux-image-extra-virtual"
+    - name: install docker and extra apt packages
+      become: yes
+      apt:
+        pkg: "{{ item }}"
+        state: present
+      with_items: "{{ apt_packages }}"
+    - name: install docker-py
+      become: yes
+      pip:
+        name: "docker-py"
+        version: "1.9.0"
+      environment:
+        LC_ALL: "C"
+    - name: add `ubuntu` to docker group
+      user:
+        append: yes
+        groups: docker
+        name: ubuntu
+      become: yes
+    - name: create ssl certs folder
+      become: yes
+      file:
+        path: "{{ ssl_certs_dir_path }}"
+        state: "directory"
+    - name: update permissions on ssl certs folder
+      become: yes
+      file:
+        path: "{{ ssl_certs_dir_path }}"
+        mode: "g=rwx,o=rwx"
+    - name: copy ssl certficate
+      become: yes
+      copy:
+        src: "{{ ssl_crt_file }}"
+        dest: "{{ ssl_certs_dir_path }}/registry-certificate.crt"
+    - name: copy ssl key
+      become: yes
+      copy:
+        src: "{{ ssl_key_file }}"
+        dest: "{{ ssl_certs_dir_path }}/registry-key.key"
+    - name: create registry directory
+      become: yes
+      file:
+        path: "{{ registry_local_dir_path }}"
+        state: directory
+        owner: ubuntu
+        group: ubuntu
+    - name: create auth directory
+      become: yes
+      file:
+        path: "{{ auth_local_dir_path }}"
+        state: directory
+        owner: ubuntu
+        group: ubuntu
+    - name: generate initial auth
+      become: yes
+      command: "docker run --entrypoint htpasswd registry:2 -Bbn {{ registry_user }} {{ registry_password }}"
+      register: auth_result
+    - name: save auth
+      copy:
+        content: "{{ auth_result.stdout }}"
+        dest: "{{ auth_local_dir_path }}/passwd"
+    - name: pull registry image
+      become: yes
+      docker_image:
+        name: registry
+        tag: 2
+        state: present
+    - name: run registry container
+      become: yes
+      docker_container:
+        name: registry
+        image: registry:2
+        pull: yes
+        restart_policy: always
+        restart: yes
+        env:
+          REGISTRY_HTTP_TLS_CERTIFICATE: "{{ ssl_certs_dir_path }}/registry-certificate.crt"
```

Comment:
cert path in the provisioner machine (your host) are provided through an env var (README on the way).